adapters/rtbhouse/rtbhouse.go

```diff
@@ -33,7 +38,51 @@ func (adapter *RTBHouseAdapter) MakeRequests(
 	requestsToBidder []*adapters.RequestData,
 	errs []error,
 ) {
-	openRTBRequestJSON, err := json.Marshal(openRTBRequest)
+
+	reqCopy := *openRTBRequest
+	reqCopy.Imp = []openrtb2.Imp{}
+	for _, imp := range openRTBRequest.Imp {
+
+		rtbhouseExt, err := getImpressionExt(imp)
```

Comment:
It seems that we want to proritize the `imp.BidFloor` over `rtbhouseExt.BidFloor` which opens the posibility of a small optimization if we do not unmarshal `rtbhouseExt` at all when the check in line 54 is `false`. In other words, if `imp.BidFloor` is greater than zero, no need to unmarshal `rtbhouseExt` at all because we'd have no use for it.
```
44     for _, imp := range openRTBRequest.Imp {
45
46 -       rtbhouseExt, err := getImpressionExt(imp)
47 -       if err != nil {
48 -           return nil, []error{err}
49 -       }
50
51         var bidFloorCur = imp.BidFloorCur
52         var bidFloor = imp.BidFloor
53
54 -       if bidFloorCur == "" && bidFloor == 0 && rtbhouseExt.BidFloor > 0 && len(reqCopy.Cur) > 0 {
   +       if bidFloorCur == "" && bidFloor == 0 {
   +           rtbhouseExt, err := getImpressionExt(imp)
   +           if err != nil {
   +               return nil, []error{err}
   +           }
   +           if rtbhouseExt.BidFloor > 0 && len(reqCopy.Cur) > 0 {
55                 bidFloorCur = reqCopy.Cur[0]
56                 bidFloor = rtbhouseExt.BidFloor
   +           }
57         }
58
adapters/rtbhouse/rtbhouse.go
```

Reply:
Hello @guscarreon , thank you for your thorough review.
You're right, we'd like to prioritize im.BidFloor over rtbhouseExt.BidFloor. There's room for optimizations as you noted and we will implement those changes.

Reply:
Changes implemented, thank you @guscarreon for your help.